### PR TITLE
Named window frames 

### LINKED ISSUE #867: Add support for named window frames
Parser cannot build a statement for named window frames. The query below can run successfully,

`SELECT SUM(amount) OVER(ORDER BY ts) as sum1 FROM sales_us;`

However, the equivalent query below with a named window frame cannot be run.

`SELECT SUM(amount) OVER running_window FROM sales_us WINDOW running_window AS (ORDER BY ts);`

Postgres supports both versions.

### PR DESCRIPTION
**Pull request by berkaysynnada**
Closes #867.

### CODE REVIEW

**Comment by mustafasrepo** [comment]
Add a unit test for named windows similar to the `parse_window_functions`.

**Review by ozankabak — APPROVED** [review]
Checked in a commit rewriting the named window search in functional style, other than that minor change it looks good to me. Let's add the test @mustafasrepo requested, expand the PR body a little bit to include an example of what this will be parsing, and then send it upstream.

File: src/parser.rs [hunk 1]
```diff
@@ -5267,6 +5247,31 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let named_windows = if self.parse_keyword(Keyword::WINDOW) {
```
Comment (berkaysynnada) [inline src/parser.rs:5250]: write a comment for this part

**Comment by mustafasrepo** [comment]
Merged upstream

**Pull request closed without merging**